src/analyses/ai_history.h

```diff
@@ -0,0 +1,237 @@
+/*******************************************************************\
+
+Module: Abstract Interpretation
+
+Author: Martin Brain, martin.brain@cs.ox.ac.uk
+
+\*******************************************************************/
+
+/// \file
+/// Abstract Interpretation history
+
+#ifndef CPROVER_ANALYSES_AI_HISTORY_H
+#define CPROVER_ANALYSES_AI_HISTORY_H
+
+#include <memory>
+
+#include <goto-programs/goto_model.h>
+
+#include <util/json.h>
+#include <util/xml.h>
+
+/// A history object is an abstraction / representation of the control
+/// control-flow part of a set of traces.  The simplest example is a single
```

Comment:
`control control-flow` .. just checking if this is intentional.

Reply:
Nope; editing error.  Fixed.